sdccc/src/test/java/com/draeger/medical/sdccc/messages/TestMessageStorage.java

```diff
@@ -2521,10 +2508,280 @@ public void testDetermineCharsetFromMessageASCIISpecialCase(@TempDir final File
                 dir, null, null, StandardCharsets.ISO_8859_1, StandardCharsets.UTF_8, "application/soap+xml");
         // then
         assertEquals(StandardCharsets.ISO_8859_1, actualCharset);
-        Mockito.verify(this.testRunObserver, VerificationModeFactory.atLeastOnce())
-                .invalidateTestRun(anyString());
+        verify(this.testRunObserver, VerificationModeFactory.atLeastOnce()).invalidateTestRun(anyString());
+    }
+
+    /**
+     * Checks that convertMessageToMessageContent() works when it does not detect any decoding problems in the message.
+     * @param dir - a temporary directory.
+     * @throws IOException - when something goes wrong.
+     */
+    @Test
+    public void testConvertToMessageContent(@TempDir final File dir) throws IOException {
+
+        final Charset charsetInHttpHeader = StandardCharsets.UTF_8;
+        final Charset charsetInXMLDeclaration = StandardCharsets.UTF_8;
+        final Charset charsetInXMLDeclarationEncoding = StandardCharsets.UTF_8;
+        final String mimeType = "application/xml";
+
+        try (final MessageStorage messageStorage = new MessageStorage(
+                1, false, true, mock(MessageFactory.class), new HibernateConfigImpl(dir), this.testRunObserver)) {
+            // given
+            final ListMultimap<String, String> headers = ArrayListMultimap.create();
+            headers.put("Content-Type", String.format("%s;charset=%s", mimeType, charsetInHttpHeader));
+
+            final HttpApplicationInfo applicationInfo = new HttpApplicationInfo(headers, "transactionId", "requestURI");
+            final TransportInfo transportInfo =
+                    new TransportInfo("http", "localhost", 1234, "remotehost", 4567, List.of());
+            final CommunicationContext communicationContext = new CommunicationContext(applicationInfo, transportInfo);
+            final Message message = new Message(
+                    CommunicationLog.Direction.INBOUND,
+                    CommunicationLog.MessageType.REQUEST,
+                    communicationContext,
+                    messageStorage);
+            final String content = String.format(
+                    "<?xml version=\"1.0\" encoding=\"%s\"?>%n<sometag></sometag>", charsetInXMLDeclaration);
+            final byte[] encodedContent = content.getBytes(charsetInXMLDeclarationEncoding);
+            message.write(encodedContent, 0, encodedContent.length);
+            message.close();
+
+            // when
+            final var result = messageStorage.convertMessageToMessageContent(message);
+
+            // then
+            assertNotNull(result);
```

Comment:
verify the content of the body field too

Reply:
Done.